### ISSUE TITLE
[FEATURE_REQUEST] start.sh improvements  and litening on custom ip

### ISSUE THREAD

**Issue by sofoxe1** [SillyTavern/SillyTavern#1409]
### Have you searched for similar requests?

Yes

### Describe the solution you'd like

1. add to start.sh npm version check, currently it only checks if npm is installed which causes it to fail with when default npm is outdated

2. nvm is used only after installing

3. ~/.nvm/nvm.sh can be sourced  instead of ~/.bashrc

4.  allow binding to specific ip as an alternative to whitelist. whitelist doesn't allow subnets, also should be easy to do

### Priority

Low (Nice-to-have)

### Is this something you would be keen to implement?

Yes!
